lib/cloudant.js

```diff
@@ -524,21 +531,52 @@ Cloudant.prototype.updateAll = function(model, where, data, options, cb) {
           model, where, data, options);
   var self = this;
   var mo = self.selectModel(model);
+  var idName = self.idName(model);
+  // if idName exists, change the idName to be the default `_id` key
+  if (idName && data[idName]) {
+    data._id = data[idName];
+    delete data[idName];
+  }
+  var res = {};
+  var docList = [];
+
   self.all(model, {where: where}, {raw: true}, function(err, docs) {
     if (err) return cb(err, docs);
     if (docs.length === 0) return cb(null, {count: 0});
 
     data = getPlainJSONData.call(self, model, data);
+
+    /**
+     * for each of the docs from the given filter, only update the instance where the revs match
+     * behavior of bulk update in cloudant:
```

Comment:
I do not quite understand what you mean.

> _rev is missing from one or more objects in data: They'll fail the rev-check in Cloudant, and will be aggregated in the list of failures returned to us in the results payload.

data is only one object to be updated. when the user specifies the `_rev`, bulk update will at most do one update to the doc where the docs match since `_rev` is **unique**.

The use case is simple:
```javascript
      Employee.create([
        {id: 1, name: 'Foo', age: 1},
        {id: 2, name: 'Bar', age: 1},
        {id: 3, name: 'Baz', age: 2},
      ],
      function(err, result) {
        if (err) throw err;
        console.log('\nCreated instance: ' + JSON.stringify(result));
        var  idInst1 = result[0].id;
        var revInst1 = result[0]._rev;

        Employee.update({age: 1}, {name: 'Sakib', age: 12, _rev: revInst1, id: idInst1},
        function(err, result) {
          if (err) throw err;
          console.log('\nUpdated instance: ' + JSON.stringify(result));

          Employee.find(function(err, result) {
            if (err) throw err;

            console.log('\nFound instance: ' + JSON.stringify(result));
          });
        });
      });
```

Feel free to explain your concern with some code.